wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -213,6 +213,7 @@
         Set<String> collidingNames = collidingFieldNames(fieldsAndOneOfFields);
         for (Field field : fieldsAndOneOfFields) {
           String suggestion = collidingNames.contains(field.getName())
+            || field.getName().equals(field.getType().getSimpleName())
```

Comment:
My apologies, the CI failure makes me realize that we didn't care about scalar types in Java because we use builders. So let's go back to your original take.
```suggestion
            || (field.getName().equals(field.getType().getSimpleName())
              && !field.getType().isScalar())
```